source/scripts/edit.js

```diff
@@ -23,4 +26,64 @@ function closeEditForm() {
   document.getElementById("editForm").style.display = "none";
 }
 
-export { openEditForm, closeEditForm };
+// -------------------------- end of edit popup window--------------------------------------
+
+// --------------------------edit button and extracting data --------------------------------------
+//when the save button in the edit form is clicked, call saveData given the ID of the particular task name
+document.getElementsByClassName("submitEditButton")[0].addEventListener("click", () => {
+  saveData(document.getElementsByClassName("submitEditButton")[0].id);
```

Comment:
I think this function could be moved inside the `openEditForm()` since that's when everything inside the edit form should be defined (correct my interpretation if wrong). If so, the submit button on the edit form would have all attributes associated declared there (id and onclick). Getting the submit element with the `id` would be feasible when adding event listener with the provided parameter, and calling `saveData(id)` would also be easier to read. 

Do note that if we want to keep this event listener here while differentiating the submit and edit button ids in `openEditForm()`, we would have to split the submit button id here to get just the `id` depending on how we format it. If we do use the syntax I pointed out above, we could parse out the `submit-` and get the latter half to pass just the hashed id into `saveData()`.

Reply:
Good advice, I have refactored the code and made it cleaner. The code is cleaner but the draw back is that we did not split up the tasks, for example there is no more `getData` function to populate the data into the edit form.